pkg/buildpod/generator.go

```diff
@@ -84,13 +91,64 @@ func (g *Generator) buildAllowed(ctx context.Context, build BuildPodable) error
 		}
 	}
 
-	for _, binding := range build.Bindings() {
-		if binding.SecretRef != nil && forbiddenSecrets[binding.SecretRef.Name] {
-			return fmt.Errorf("binding %q uses forbidden secret %q", binding.Name, binding.SecretRef.Name)
+	bindings := make([]buildapi.ServiceBinding, 0)
+
+	if services := build.Services(); len(services) != 0 {
+		for _, s := range build.Services() {
+			var b corev1alpha1.ServiceBinding
+			if s.Kind == "Secret" {
+				b = corev1alpha1.ServiceBinding{
+					Name:      s.Name,
+					SecretRef: &corev1.LocalObjectReference{Name: s.Name},
+				}
+			} else {
+				gvr, _ := meta.UnsafeGuessKindToResource(s.GroupVersionKind())
```

Comment:
Can we extract this to provisonedService, err := readProvisonedServiceDuckType()